diesel_codegen/src/schema_inference.rs

```diff
@@ -14,15 +14,12 @@ pub fn derive_infer_schema(input: syn::MacroInput) -> quote::Tokens {
     let options = get_options_from_input(&input.attrs, bug).unwrap_or_else(|| bug());
     let database_url = get_option(&options, "database_url", bug);
 
-    let table_names = load_table_names(&database_url).unwrap();
-    let schema_inferences = table_names.into_iter().map(|table_name| {
-        let mod_ident = syn::Ident::new(format!("infer_{}", table_name));
-        quote! {
-            mod #mod_ident {
-                infer_table_from_schema!(#database_url, #table_name);
-            }
-            pub use self::#mod_ident::*;
-        }
+    let schema_names = load_schema_names(&database_url).unwrap()
+        .into_iter()
+        .map(Some)
+        .chain(vec![None]);
```

Comment:
The last item is `None` to default to loading the `public` schema, right? Might be nice to add a comment :)